### ISSUE TITLE
Union with a record

### ISSUE THREAD

**Issue by untereiner** [lerouxrgd/rsgen-avro#41]
Hello
I try to generate rust code form a set of avsc files.
I get an error. Here is a minimal example generating the same error:

> Templating error: Requires Schema::Union, found Record { name: Name { name: "B", namespace: None }, aliases: None, doc: None, fields: [RecordField { name: "pass", doc: None, default: None, schema: Long, order: Ascending, position: 0 }, RecordField { name: "direction", doc: None, default: None, schema: Ref { name: Name { name: "C", namespace: None } }, order: Ascending, position: 1 }, RecordField { name: "depth", doc: None, default: None, schema: Double, order: Ascending, position: 2 }], lookup: {"depth": 2, "direction": 1, "pass": 0} }

I think the problem is line 662 in templates.rs. I guess a reference to another schema should not only return a str_union_enum ?
 

```
{
	"type": "record",
	"name": "A",
	"fields":
	[
		{"name": "item",
			"type": [
				"null",
				"long",
				"double",
				"B"
			]
		}
	]
}
{
	"type": "record",
	"name": "B",
	"fields":
	[
		{ "name": "pass", "type": "long" },
		{ "name": "direction", "type": "C" },
		{ "name": "depth", "type": "double" }
	]
}
{
	"type": "enum",
	"name": "C",
	"symbols":
	[
		"U",
		"S",
		"D"
	]
}
```

**Comment by lerouxrgd** [comment]
Thanks for narrowing it down ! It has been fixed in `0.11.4`, let me know how it goes.

**Comment by untereiner** [comment]
yes it works ! Thank you !
